packages/vendure-plugin-sort-by-popularity/src/sort.service.ts

```diff
@@ -1,12 +1,83 @@
 import { Injectable } from '@nestjs/common';
-
+import {
+  Collection,
+  CollectionService,
+  OrderLine,
+  Product,
+  RequestContext,
+  TransactionalConnection,
+} from '@vendure/core';
+import { Success } from '../../test/src/generated/admin-graphql';
+import { CollectionTreeNode } from './helpers';
 @Injectable()
 export class SortService {
-  async setProductPopularity() {
-    // Get orders from pas X months
-    // Combine variant quantities per product
-    // Normalize by total quantity and store as popularity score on product
-    // Get all products and products of child collections for each collection
-    // Use sum of products per collection as popularity score
+  constructor(
+    private connection: TransactionalConnection,
+    private collectionService: CollectionService
+  ) {}
+  async setProductPopularity(ctx: RequestContext): Promise<Success> {
```

Comment:
We can do this in another PR, but:
We should make this endpoint push a job to the worker. There is an example here: https://github.com/Pinelab-studio/pinelab-vendure-plugins/blob/f45644e3f39bd7eef948e930b91dde5f823b703a/packages/vendure-plugin-sendcloud/src/api/sendcloud.service.ts#L83

Reply:
Ok I will use the worker for that